starcluster/cluster.py

```diff
@@ -1213,20 +1213,28 @@ def create_nodes(self, aliases, image_id=None, instance_type=None,
             log.info(str(resv), extra=dict(__raw__=True))
         return resvs
 
-    def _get_next_node_num(self):
+    @classmethod
+    def get_free_ids_among_nodes(cls, count, nodes):
+        ids = [int(re.search('node(\d{3})', n.alias).group(1)) for n in nodes]
+        print "ids", ids
+        result = []
+        remaining = count
+        for i in xrange(1, 1000):
+            if i not in ids:
+                result.append(i)
+                remaining -= 1
+            if remaining == 0:
```

Comment:
nitpick: if that function is called with count=0, it will loop until the end of the range. Using <= count would avoid that.